apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Query.cs

```diff
@@ -493,10 +496,9 @@ private Query Where(FieldPath fieldPath, FieldOp op, object value)
                 };
             }
 
-            InternalFilter filter = InternalFilter.Create(Database.SerializationContext, fieldPath, op, value);
-            var newFilters = _filters == null ? new List<InternalFilter>() : new List<InternalFilter>(_filters);
-            newFilters.Add(filter);
-            return new Query(_root, _offset, _limit, _orderings, newFilters, _projections, _startAt, _endAt);
+            var newFilter = (_filter == null) ? new Filter.CompositeFilter(new List<Filter>(), CompositeFilter.Types.Operator.And)
```

Comment:
Any reason this isn't just `return Where(new Filter.UnaryFilter(fieldPath, op, value))`? (I suspect eventually we'll move all of this logic somewhere into Filter anyway.)

Reply:
yups that is the eventually what the aim right now all the things are not in-place so its like this.